include/swift/Serialization/SerializedModuleLoader.h

```diff
@@ -64,9 +64,11 @@ class SerializedModuleLoaderBase : public ModuleLoader {
   ASTContext &Ctx;
   ModuleLoadingMode LoadMode;
   bool IgnoreSwiftSourceInfoFile;
+  bool ImplicitModulesDisabled;
```

Comment:
Instead of passing down whether implicit module is disabled, can we avoid adding `SerializedModuleLoader` at all for the option is set?

Reply:
Meanwhile, we could rename `SerializedModuleLoader` to `ImplicitSerializedModuleLoader`.

Reply:
I tried that first, but ran into problems because we use the SerializedModuleLoader to load partial modules in `CompilerInstance::loadPartialModulesAndImplicitImports`, so it always needs to be present.

I will try and see if I can implement your suggestion of refactoring it into an `ImplicitSerializedModuleLoader `, and use a different loader for partial modules.

Reply:
Thanks! Partial modules are closer to explicit modules, right? we could move the logic to `ExplicitModuleLoader` and use that for partial module loading.

Reply:
Yes, they are, in the sense that they are direct inputs on the command line. 👍

Reply:
Yeah, potentially we can move partial modules into using the JSON format as well like other explicit modules, but it can come later.